server/render.js

```diff
@@ -11,33 +12,55 @@ import serialize from 'serialize-javascript';
 import routes from '../app/routes';
 import configureStore from '../app/utils/configureStore';
 import config from '../config/env';
+import type { $Request, $Response, Middleware } from 'express';
+import { UniversalRavenNode } from '../app/utils/universalRaven';
+import webpackClient from '../config/webpack.client.js';
+import type { State } from '../app/types';
 
 import manifest from '../app/assets/manifest.json';
 
 const serverSideTimeoutInMs = 4000;
 
-const TimeoutError = new Error(
-  'React prepare timeout when server side rendering.'
-);
+// AntiPattern because of babel
+// https://github.com/babel/babel/issues/3083
+class TimeoutError {
+  error: Error;
+  constructor(msg) {
+    this.error = new Error(msg);
+  }
+}
 
-function prepareWithTimeout(app) {
-  return Promise.race([
+const prepareWithTimeout = app =>
+  Promise.race([
     prepare(app),
     new Promise(resolve => {
       setTimeout(resolve, serverSideTimeoutInMs);
     }).then(() => {
-      throw TimeoutError;
+      throw new TimeoutError(
+        'React prepare timeout when server side rendering.'
+      );
     })
   ]);
-}
 
-function render(req, res, next) {
+function render(req: $Request, res: $Response, next: Middleware) {
+  const render = (
+    body: string = '',
+    state: State | {||} = Object.freeze({})
```

Comment:
The sealed empty object looked like a good idea, but this `Object.freeze` looks aweful :laughing: